language/stdlib/modules/AccountLimits.move

```diff
@@ -165,26 +156,38 @@ module AccountLimits {
         new_max_inflow: u64,
         new_max_outflow: u64,
         new_max_holding_balance: u64,
+        new_time_period: u64,
     ) acquires LimitsDefinition {
-        assert(has_treasury_compliance_role(tc_account), ENOT_TREASURY_COMPLIANCE);
+        assert(Roles::has_treasury_compliance_role(tc_account), ENOT_TREASURY_COMPLIANCE);
         // As we don't have Optionals for txn scripts, in update_account_limit_definition.move
         // we use 0 value to represent a None (ie no update to that variable)
         let limits_def = borrow_global_mut<LimitsDefinition<CoinType>>(limit_address);
         if (new_max_inflow > 0) { limits_def.max_inflow = new_max_inflow };
         if (new_max_outflow > 0) { limits_def.max_outflow = new_max_outflow };
         if (new_max_holding_balance > 0) { limits_def.max_holding = new_max_holding_balance };
+        if (new_time_period > 0) { limits_def.time_period = new_time_period };
     }
 
-    /// Since we don't track balances of accounts before they are limited, once
-    /// they do become limited the approximate balance in `CointType` held by
-    /// the entity across all of its accounts will need to be set by the association.
-    public fun set_current_holdings<CoinType>(
+    /// Update either the `tracked_balance` or `limit_address` fields of the
+    /// `Window<CoinType>` stored under `window_address`.
+    /// * Since we don't track balances of accounts before they are limited, once
+    ///   they do become limited the approximate balance in `CointType` held by
+    ///   the entity across all of its accounts will need to be set by the association.
+    ///   if `aggregate_balance` is set to zero the field is not updated.
+    /// * If we wish to point this window to a new limits definition, we can do
+    ///   so by passing a `new_limit_address` to this function. If the
```

Comment:
Are we guaranteeing that new_limit_address has a LimitsDefinition resource via genesis?
Why is VM_RESERVED_ADDRESS specifically used as a means of bypassing setting the new_limit_address?

Reply:
Good questions. 
* You are not guaranteed that the limits definition resource is published in genesis. You can see an example of publishing a limits definition after genesis in the updates to the `account_limits` test. The association will need to make sure that the limits definition resource lives at the address before pointing the window to it. Windows are by default pointed to global limits definitions when they are created, so they will remain consistent. I'll add a check to make sure a limits definition live at the `new_limit_address` address.
* We use VM_RESERVED_ADDRESS since no one can send transactions from that address except for the VM, in particular, no account can be created there. Because of this, short of the VM creating a limits definition at its address (highly highly unlikely) this address will never hold a limits definition resource so it's unlikely you'll want to set a new_limit_address to point at the VM's reserved address.